src/fork_gc.c

```diff
@@ -777,6 +777,7 @@ static FGCError FGC_parentHandleTerms(ForkGC *gc, RedisModuleCtx *rctx) {
       dictDelete(sctx->spec->keysDict, termKey);
     }
     Trie_Delete(sctx->spec->terms, term, len);
+    sctx->spec->stats.numTerms--;
```

Comment:
During addition, also the `termsSize` field was [incremented](https://github.com/RediSearch/RediSearch/blob/39a46e196b4f313ca57dea965b9dfcc3166e580f/src/spec.c#L977) - shouldn't be adjusted on deletion?

Reply:
Done. Good catch.